php/libraries/Utility.class.inc

```diff
@@ -558,7 +558,7 @@ class Utility
      */
     static function asArray($var)
     {
-        if (!is_array($var)) {
+        if (!is_array($var) && !empty($var)) {
```

Comment:
@driusan  this still should be changed though otherwise you return `array([0] => )`

dont you think ?